python/ray/air/execution/tracked_actor.py

```diff
@@ -0,0 +1,31 @@
+from typing import Callable
+
+
+class TrackedActor:
+    """Actor tracked by an actor manager.
+
+    This object is used to reference a Ray actor on an actor manager
+
+    Existence of this object does not mean that the Ray actor has already been started.
+    Actor state can be inquired from the actor manager tracking the Ray actor.
+
+    Attributes:
+        actor_id: ID for identification of the actor within the actor manager. This
+            ID is not related to the Ray actor ID.
+
+    """
+
+    def __init__(self, actor_id: int):
+        self.actor_id = actor_id
+
+    def on_start(self, callback: Callable[["TrackedActor"], None]) -> "TrackedActor":
+        """Set callback to invoke when actor started."""
+        raise NotImplementedError
+
+    def on_stop(self, callback: Callable[["TrackedActor"], None]) -> "TrackedActor":
+        """Set callback to invoke when actor stopped gracefully."""
+        raise NotImplementedError
+
+    def on_error(self, callback: Callable[["TrackedActor"], None]) -> "TrackedActor":
+        """Set callback to invoke when actor died."""
```

Comment:
May want to rename these methods to set_*, to make explicit that these are only setting the callbacks, but are not actually executing it.

Speaking of which, is there a separate API to actually trigger the callbacks?

Reply:
IMO the `on_` API is pretty clear about not executing the callbacks: `future.on_error(handle_error).on_timeout(handle_timeout)` etc - why would anyone assume the callbacks are executed immediately?

The callbacks will be triggered whenever the event arrives. E.g. when the future resolves, the `on_result` callback will be called. When the future errors, the `on_error` callback will be called.

For this we have to yield control to the event manager via `manager.wait()`. The alternative would be to do this in a separate thread, but then we run into problems with thread-safety, so I think for now that's the best way to go forward.